crates/net/network/src/receiver.rs

```diff
@@ -0,0 +1,32 @@
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use futures::Future;
+use tokio::sync::oneshot::{error::RecvError, Receiver};
+
+pub struct ResolvedOneshotReceiver<T> {
```

Comment:
this needs `Debug` and some docs.

I'd also like to rename this.

this is basically behaves like `Result::flatten`, so perhaps `FlattenedResponse`?